### PR TITLE
fix(core): remove obsolete utils

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
## Related Issue(s)

n/a

## Description
In core/utils there're a few functions that are no longer needed:
- cloneDeep has been replaced with lodash's implementation. Current implementation could not copy classes properly. 
- `compareObjects` has been replaced with `fast-deep-equal`, because it's the fastest object comparer
- `closestElement` has been removed in favour of `Element.closest()` method, that is supported by all browsers except IE (which we're not targeting anymore)
- `getNestedValue` has been replaced with lodash's `get`
- `mergeDeep` has been replaced with lodash's `merge`

Each change is shipped as a separate commit

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [n/a] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Comment by dimamarksman** [comment]
I would suggest keeping these utils anyway.
If we want to update the implementation of these helpers we can update just it, so imports and interfaces remain the same.
In this case we don't need so many changes in code.

@dmitry-stepanenko what do you think?

**Comment by dmitry-stepanenko** [comment]
@dimamarksman in my opinion it will be quite messy to keep functions that simply wraps other functions without any need. What are the benefits of this? Moreover, `cloneDeep` and `fast-deep-equal` even have the same interfaces

**Comment by dimamarksman** [comment]
@dmitry-stepanenko it gives us encapsulation and strict interfaces for consumers so end consumer (components) doesn't know about third paty libs.
In the future it gives us ability to update helpers implementation without changing 10+ files.
Of course if we can rely on native API we don't need any helper at all in this case

**Comment by dmitry-stepanenko** [comment]
@dimamarksman as discussed, retained the current implementation

File: apps/docs/src/app/platform/component-docs/platform-icon-tab-bar/examples/platform-icon-tab-bar-configurable-paddings-example/platform-icon-tab-bar-configurable-paddings-example.component.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Component } from '@angular/core';
-import { cloneDeep } from '@fundamental-ngx/core';
+import { cloneDeep } from 'lodash-es';
```
Comment (N1XUS) [inline apps/docs/src/app/platform/component-docs/platform-icon-tab-bar/examples/platform-icon-tab-bar-configurable-paddings-example/platform-icon-tab-bar-configurable-paddings-example.component.ts:2]: please use deep import from lodash-es/cloneDeep

Comment (N1XUS) [reply apps/docs/src/app/platform/component-docs/platform-icon-tab-bar/examples/platform-icon-tab-bar-configurable-paddings-example/platform-icon-tab-bar-configurable-paddings-example.component.ts:2]: I would also update blacklist of imports to restrict import from the root of lodash

File: libs/platform/src/lib/table/table.component.ts [hunk 2]
```diff
@@ -35,8 +35,9 @@ import {
     RtlService
 } from '@fundamental-ngx/core/utils';
 import { TableRowDirective } from '@fundamental-ngx/core/table';
-import { getNestedValue, isDataSource, isFunction, isString } from '@fundamental-ngx/platform/shared';
+import { isDataSource, isFunction, isString } from '@fundamental-ngx/platform/shared';
 import { PopoverComponent } from '@fundamental-ngx/core/popover';
+import { get } from 'lodash';
```
Comment (N1XUS) [inline libs/platform/src/lib/table/table.component.ts:40]: maybe lodash-es?

File: package.json [hunk 3]
```diff
@@ -128,6 +128,7 @@
         "intl": "1.2.5",
         "jasminewd2": "2.2.0",
         "karma-viewport": "1.0.8",
+        "lodash-es": "4.17.21",
```
Comment (N1XUS) [inline package.json:131]: Please also update subpackages package.json files to reflect this dependency

**Review by N1XUS — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/table/pipes/value-by-path.pipe.ts [hunk 4]
```diff
@@ -1,10 +1,9 @@
 import { Pipe, PipeTransform } from '@angular/core';
-
-import { getNestedValue } from '@fundamental-ngx/platform/shared';
+import { get } from 'lodash';
```
Comment (dimamarksman) [inline libs/platform/src/lib/table/pipes/value-by-path.pipe.ts:2]: lodash-es?

File: apps/docs/src/app/platform/component-docs/platform-icon-tab-bar/examples/platform-icon-tab-bar-configurable-paddings-example/platform-icon-tab-bar-configurable-paddings-example.component.ts (hunk 1 above)
Comment (dmitry-stepanenko) [reply apps/docs/src/app/platform/component-docs/platform-icon-tab-bar/examples/platform-icon-tab-bar-configurable-paddings-example/platform-icon-tab-bar-configurable-paddings-example.component.ts:2]: Tree-shaking works fine, when dependencies are imported from `lodash-es` directly. I've verified.

File: libs/platform/src/lib/table/pipes/value-by-path.pipe.ts (hunk 4 above)
Comment (dmitry-stepanenko) [reply libs/platform/src/lib/table/pipes/value-by-path.pipe.ts:2]: fixed

File: apps/docs/src/app/platform/component-docs/platform-icon-tab-bar/examples/platform-icon-tab-bar-configurable-paddings-example/platform-icon-tab-bar-configurable-paddings-example.component.ts (hunk 1 above)
Comment (dmitry-stepanenko) [reply apps/docs/src/app/platform/component-docs/platform-icon-tab-bar/examples/platform-icon-tab-bar-configurable-paddings-example/platform-icon-tab-bar-configurable-paddings-example.component.ts:2]: > I would also update blacklist of imports to restrict import from the root of lodash
This is not easily doable at this point, I've created a separate ticket
https://github.com/SAP/fundamental-ngx/issues/7414

File: package.json (hunk 3 above)
Comment (dmitry-stepanenko) [reply package.json:131]: done

**Review by dimamarksman — APPROVED** [review]

**Review by N1XUS — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request merged**